browser/ipfs/ipfs_service_delegate_impl.h

```diff
@@ -0,0 +1,24 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_IPFS_IPFS_SERVICE_DELEGATE_IMPL_H_
+#define BRAVE_BROWSER_IPFS_IPFS_SERVICE_DELEGATE_IMPL_H_
+
+#include "brave/components/ipfs/browser/ipfs_service_delegate.h"
+
+namespace ipfs {
+
+class IpfsServiceDelegateImpl : public IpfsServiceDelegate {
```

Comment:
btw, it appears to me that the typical chromium pattern is to name base interface as "client" (i.e. `IpfsServiceClient`) and its child class as "Chrome...Client" (since the embedder is Chrome, it could also be android). So in our case it probably should be `BraveIpfsServiceClient` and `IpfsServiceClient`.

Reply:
I see quite a lot of uses of Delegate...

```
browser/autocomplete/keyword_extensions_delegate_impl.h
browser/background_fetch/background_fetch_delegate_impl.h
browser/chromeos/arc/session/adb_sideloading_availability_delegate_impl.h
browser/chromeos/display/quirks_manager_delegate_impl.h
browser/chromeos/events/event_rewriter_delegate_impl.h
browser/chromeos/input_method/input_method_delegate_impl.h
browser/chromeos/policy/minimum_version_policy_handler_delegate_impl.h
browser/chromeos/printing/cups_proxy_service_delegate_impl.h
browser/extensions/api/passwords_private/passwords_private_delegate_impl.h
browser/lacros/lacros_chrome_service_delegate_impl.h
browser/media/router/presentation/presentation_service_delegate_impl.h
browser/media/router/presentation/receiver_presentation_service_delegate_impl.h
browser/nacl_host/nacl_browser_delegate_impl.h
browser/safe_browsing/url_checker_delegate_impl.h
browser/signin/header_modification_delegate_impl.h
browser/signin/process_dice_header_delegate_impl.h
browser/speech/tts_controller_delegate_impl.h
browser/supervised_user/supervised_user_extensions_delegate_impl.h
browser/ui/ash/assistant/device_actions_delegate_impl.h
browser/ui/sync/one_click_signin_links_delegate_impl.h
browser/ui/webui/signin/dice_turn_sync_on_helper_delegate_impl.h
renderer/supervised_user/supervised_user_error_page_controller_delegate_impl.h
```

Reply:
yeah, I've double-checked: there are quite many delegates (to be more representative, check `chrome_*_delegate.h` and `chrome_*_client.h`. It seems the rule of thumb is to call a fat interface `Client` and something more specific `Delegate`